nuxeo-coldstorage/src/test/java/org/nuxeo/coldstorage/operations/TestColdStorageRendition.java

```diff
@@ -0,0 +1,181 @@
+
+        Blob mainContent = (Blob) document.getPropertyValue(ColdStorageConstants.FILE_CONTENT_PROPERTY);
+
+        // Check if the FILE_CONTENT_PROPERTY and the rendition are equals
+        assertEquals(expectedBlob.getString(), mainContent.getString());
+    }
+
+    @Test
+    public void shouldSelectSmallRendition() throws IOException, OperationException {
+        DocumentModel document = createPictureDocumentModel(session);
+
+        transactionalFeature.nextTransaction();
+
+        document.refresh();
+
+        // Retrieve the expected rendition
+        RenditionService renditionService = Framework.getService(RenditionService.class);
+        Rendition rendition = renditionService.getRendition(document, "Small");
+        Blob expectedBlob = rendition.getBlob();
+
+        moveContentToColdStorage(session, document);
+
+        transactionalFeature.nextTransaction();
+        document.refresh();
+
+        Blob mainContent = (Blob) document.getPropertyValue(ColdStorageConstants.FILE_CONTENT_PROPERTY);
+
+        // Check if the FILE_CONTENT_PROPERTY and the rendition are equals
+        assertEquals(expectedBlob.getString(), mainContent.getString());
+    }
+
+    @Test
+    @Deploy("org.nuxeo.coldstorage.test:OSGI-INF/test-coldstorage-rendition-contrib.xml")
+    public void shouldFailRenditionNotExist() throws IOException, OperationException {
+        DocumentModel document = createPictureDocumentModel(session);
+
+        transactionalFeature.nextTransaction();
+        document.refresh();
+
+        try {
+            moveContentToColdStorage(session, document);
+        } catch (NuxeoException e) {
+            assertEquals(SC_NOT_FOUND, e.getStatusCode());
+        }
+    }
+
+    @Test
+    public void shouldSelectDefaultVideoRendition() throws IOException, OperationException {
+        DocumentModel document = createVideoDocumentModel(session, "images/sample.mpg");
+
+        transactionalFeature.nextTransaction();
+
+        document.refresh();
+
+        // Retrieve the expected rendition
+        RenditionService renditionService = Framework.getService(RenditionService.class);
+        Rendition rendition = renditionService.getRendition(document, "MP4 480p");
+        Blob expectedBlob = rendition.getBlob();
+
+        moveContentToColdStorage(session, document);
+
+        transactionalFeature.nextTransaction();
+        document.refresh();
+
+        Blob mainContent = (Blob) document.getPropertyValue(ColdStorageConstants.FILE_CONTENT_PROPERTY);
+
+        // Check if the FILE_CONTENT_PROPERTY and the rendition are equals
+        assertEquals(expectedBlob.getString(), mainContent.getString());
+    }
+
+    private DocumentModel createPictureDocumentModel(CoreSession session) throws IOException {
+        DocumentModel document = session.createDocumentModel("/", "picture", "Picture");
+        Blob blob = Blobs.createBlob(FileUtils.getResourceFileFromContext("images/test.jpg"), "image/jpeg",
```

Comment:
Not sure the encoding is relevant for a Picture, you could probably leave it `null`.